src/index.js

```diff
@@ -1,53 +1,189 @@
+  oldItem.replaceWith(newInput);
+  const local = localStorage.getItem('items');
+  const data = JSON.parse(local);
+  newInput.addEventListener('keypress', (e) => {
+    const newId = newInput.parentElement.id;
+    if (e.key === 'Enter') {
+      const oldItem = document.createElement('div');
+      oldItem.textContent = newInput.value;
+      newInput.replaceWith(oldItem);
+      const editedContainers = document.querySelectorAll('.new-item');
+      for (let i = 0; i < editedContainers.length; i += 1) {
+        if ((editedContainers[i].id) === newId) {
+          data[i].description = newInput.value;
+          localStorage.setItem('items', JSON.stringify(data));
+        }
+      }
+    }
+  });
+};
+
+// Remove items
+const removeItems = (li) => {
+  listOfItems.removeChild(li);
+  let count = 0;
+  const parsedItems = localStorage.getItem('items');
+  let localData = JSON.parse(parsedItems);
+  // Filter true elements
+  localData = localData.filter((item) => item.completed === false);
+  // Update index of elements
+  localData.map((item) => {
+    item.index = count;
+    count += 1;
+    return null;
+  });
+  localStorage.setItem('items', JSON.stringify(localData));
+};
+
+// Add new item
+const addNewItem = (newDescription) => {
+  const newItem = document.createElement('div');
+  newItem.classList.add('list');
+  newItem.innerHTML += `
+<input type="checkbox" class="checkbox">
+<span>${newDescription}</span>
+<span class="ellipsis">
+<i class="fa-solid fa-ellipsis-vertical"></i>
+</span>
+<span class="trashcan">
+<i class="fa-solid fa-trash-can"></i>
+</span>
 `;
+  listOfItems.appendChild(newItem);
+  const checkbox = document.querySelectorAll('.checkbox');
+  checkbox.forEach((checkboxInput) => {
+    checkboxInput.addEventListener('click', () => {
+      checkboxInput.nextElementSibling.classList.toggle('strike');
+      checkboxInput.parentNode.classList.toggle('clicked-on');
+      const ellipsisIcon = checkboxInput.parentNode.childNodes[5];
+      ellipsisIcon.classList.toggle('inactive-ellipsis');
+      const trashIcon = checkboxInput.parentNode.childNodes[7].childNodes[1];
+      trashIcon.classList.toggle('active-trash');
+      completeItem();
+    });
+  });
+  // Create and send new item to local storage
+  const newListItem = new ListItem(newDescription, false, checkbox.length - 1);
+  itemArray.push(newListItem);
+  const stringedItems = JSON.stringify(itemArray);
+  localStorage.setItem('items', stringedItems);
+  // Edit items
+  const editItem = document.querySelectorAll('.ellipsis');
+  editItem.forEach((item) => {
+    item.addEventListener('click', () => {
+      item.parentNode.classList.add('clicked-on');
+      editItems(item.previousElementSibling);
+    });
+  });
+  // Remove items
+  const removeItem = document.querySelectorAll('.trashcan');
+  removeItem.forEach((item) => {
+    item.addEventListener('click', () => {
+      removeItems(item.parentNode);
+    });
+  });
+};
```

Comment:
- [ ] Great work implementing the function that removes a single task. However, I noticed a bug. When I remove a single task from the array, it successfully deletes it from the item from the array and the local storage. But if I add a task immediately after deleting the former item, it renders the new task along with the deleted one. A quick fix would be to immediately reload the page after saving the current data to the `localStorage` in `removeItems` function. 👍